js/packages/web/src/components/AuctionCard/index.tsx

```diff
@@ -516,74 +520,132 @@ export const AuctionCard = ({
-                if (
-                  instantSalePrice &&
-                  (allowBidToPublic || allowBidToAuctionOwner)
-                ) {
+
+                try {
+                  const instantSalePrice =
+                    auctionView.auctionDataExtended?.info.instantSalePrice;
+                  const winningConfigType =
+                    auctionView.items[0][0].winningConfigType;
+                  const isAuctionItemMaster =
+                    winningConfigType ===
+                      WinningConfigType.FullRightsTransfer ||
+                    winningConfigType === WinningConfigType.TokenOnlyTransfer;
+                  const allowBidToPublic =
+                    myPayingAccount &&
+                    !auctionView.myBidderPot &&
+                    isAuctionManagerAuthorityNotWalletOwner;
+                  const allowBidToAuctionOwner =
+                    myPayingAccount &&
+                    !isAuctionManagerAuthorityNotWalletOwner &&
+                    isAuctionItemMaster;
+
+                  // Placing a "bid" of the full amount results in a purchase to redeem.
+                  if (
+                    instantSalePrice &&
+                    (allowBidToPublic || allowBidToAuctionOwner)
+                  ) {
+                    let bidTxid: string | undefined;
+
+                    try {
+                      const { amount, txid } = await sendPlaceBid(
+                        connection,
+                        wallet,
+                        myPayingAccount.pubkey,
+                        auctionView,
+                        accountByMint,
+                        instantSalePrice,
+                        'confirmed',
+                      );
+                      setLastBid({ amount });
+                      bidTxid = txid;
+                    } catch (e) {
+                      console.error('sendPlaceBid', e);
+                      return;
+                    }
+
+                    try {
+                      // Attempt to load the transaction, retrying up to 5 times
+                      retry: do {
+                        for (let i = 0; i < 5; ++i) {
+                          const tx = await connection.getTransaction(bidTxid, {
+                            commitment: 'confirmed',
+                          });
+
+                          const keys = tx?.transaction.message.accountKeys;
+
+                          if (!keys) {
+                            await new Promise(o => setTimeout(o, 2000));
+                            continue;
+                          }
+
+                          const patch = await loadMultipleAccounts(
+                            connection,
+                            keys.map(k => k.toBase58()),
+                            'confirmed',
+                          );
+
+                          const newState = patchState(patch);
+                          setShowBidModal(true);
+                          setLoading(true);
```

Comment:
This is an attempt to keep the modal up. It was observed during testing the modal would disappear and then re-apper.